### PR TITLE
Plans: Auto Redirect to checkout for premium

### PR DESCRIPTION
**Pull request by artpi**
# Problem

Currently in the checkout flow, there are following (amongst others)  steps:
1. Choose a plan
2. Choose login, pwd, email
3. Wait 8+ seconds while watching "we're setting up your site"
4. CLICK A BUTTON WHILE BEING DISTRACTED BY ENCOURAGEMENT TO CHECK YOUR EMAIL
5. Provide CC info to actually give us money

# Solution

This PR removes 5 if you have selected a plan
[Video demo](https://cloudup.com/cSZSnYmyGQD)

# Testing 

1. Use incognito (to be a new user)
2. Assign yourself signupCheckoutRedirect:auto test
3. Go through checkout, selecting a free plan
4. See that behaviour is not changed
5. Go through checkout, selecting any plan
6. See that button "clicks itself"

CC @rralian @lamosty @gwwar @shaunandrews

### CODE REVIEW

**Comment by lamosty** [comment]
Sorry, I don't understand the title. :) Is this related to signup checkout flow as you said in #7515?

**Comment by artpi** [comment]
> Sorry, I don't understand the title

It was Work in progress pr, updated now :)

**Comment by gwwar** [comment]
1. With an existing user, navigate to: http://calypso.localhost:3000/start
2. Follow site creation steps until you reach plans
3. Select a free plan
4. Wait for the loader
5. We don't progress to the next step automatically ( visiting your newly created site )

**Comment by gwwar** [comment]
👍 The described flows are looking good in testing

File: client/signup/main.jsx [hunk 1]
```diff
@@ -45,7 +45,8 @@ import * as oauthToken from 'lib/oauth-token';
 /**
  * Constants
  */
-const MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED = 8000;
+const MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED_FREE = 8000;
+const MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED_PREMIUM = 3000;
```
Comment (lamosty) [inline client/signup/main.jsx]: Not sure if users will be able to read all the text (mainly the email confirmation text) in three seconds. I would give it at least 4-5 seconds. Also, is the "waiting page" fully translated? If not, we should definitely increase the wait time as non-native speakers need more time for reading English. :)

Comment (lamosty) [inline client/signup/main.jsx]: After reading p3Ex-1tC-p2, I think three seconds are enough as we don't want the users to click on the "confirm email" link before going to checkout.

Comment (rralian) [inline client/signup/main.jsx]: IMHO we shouldn't worry the user about email confirmation at all at this step. I saw that Delta had proposed an in-context notice about email confirmation in the calypso editor right by the publish button, where it's most relevant (@meremagee). Once we drop the email confirmation text (in a future PR) we should drop this wait time as well.

**Comment by rralian** [comment]
Let's run this at 50/50 with a hypothesis of impacting checkout conversion by 10%. This should take us about one week to run. I don't actually think we'll get that lift, but we'll get some indication of whether this is beneficial and worth optimizing further. Mostly I'm curious to better understand the impact of the current behavior. 🚢

File: client/lib/abtest/active-tests.js [hunk 2]
```diff
@@ -41,6 +41,14 @@ module.exports = {
 		},
 		defaultVariation: 'singlePurchaseFlow'
 	},
+	signupCheckoutRedirect: {
+		datestamp: '20160901',
```
Comment (rralian) [inline client/lib/abtest/active-tests.js]: Don't forget to update this prior to launch.

**Pull request merged**